sequencer/src/block/namespace_payload/iter.rs

```diff
@@ -0,0 +1,77 @@
+use crate::block::{
+    full_payload::{NsIndex, NsIter, Payload},
+    namespace_payload::types::{TxIndex, TxIter},
+};
+use serde::{Deserialize, Serialize};
+use std::iter::Peekable;
+
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct Index {
+    ns_index: NsIndex,
+    tx_index: TxIndex,
+}
+
+impl Index {
+    pub fn ns(&self) -> &NsIndex {
+        &self.ns_index
+    }
+    pub(in crate::block) fn tx(&self) -> &TxIndex {
+        &self.tx_index
+    }
+}
+
+// TODO don't impl `PartialOrd`
+impl PartialOrd for Index {
+    fn partial_cmp(&self, _other: &Self) -> Option<std::cmp::Ordering> {
+        Some(self.cmp(_other))
+    }
+}
+// TODO don't impl `Ord`
+impl Ord for Index {
+    fn cmp(&self, _other: &Self) -> std::cmp::Ordering {
+        unimplemented!()
+    }
+}
+
+/// Cartesian product of [`NsIter`], [`TxIter`].
+pub struct Iter<'a> {
+    ns_iter: Peekable<NsIter<'a>>,
+    tx_iter: Option<TxIter>,
+    block: &'a Payload,
+}
+
+impl<'a> Iter<'a> {
+    pub fn new(block: &'a Payload) -> Self {
+        Self {
+            ns_iter: NsIter::new(block.ns_table()).peekable(),
+            tx_iter: None,
+            block,
+        }
+    }
+}
+
+impl<'a> Iterator for Iter<'a> {
+    type Item = Index;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            let Some(ns_index) = self.ns_iter.peek() else {
+                return None; // ns_iter consumed
```

Comment:
nit: other places you've used `break None`.

Reply:
fixed in https://github.com/EspressoSystems/espresso-sequencer/pull/1499/commits/a1269faf9d827acf444390f46c71fa52a45ad532